alibi_detect/saving/_tensorflow/loading.py

```diff
@@ -69,7 +69,7 @@ def load_model(filepath: Union[str, os.PathLike],
     return model
 
 
-def prep_model_and_emb(model: Optional[Callable], emb: Optional[TransformerEmbedding]) -> Callable:
+def prep_model_and_emb(model: Callable, emb: Optional[TransformerEmbedding]) -> Callable:
```

Comment:
`model` no longer `Optional`, since the logic [here](https://github.com/SeldonIO/alibi-detect/pull/642/files#diff-1da3b4e01d459b964951c3287ce414a19a433192bb74ea41bc4520f026f7b9dfR226-R231) is changed so that `prep_model_and_emb` is not called if `model is None`.

Reply:
I'm not sure we can use `SupportedModelsType` as is, since it contains if-statements conditional on the installed optional deps, so I don't think a static type checker (mypy) will like it? I wonder if we could do something clever by combining it with a `Protocol` (https://www.madelyneriksen.com/blog/validated-container-types-python-pydantic)? However, I've just realised that `UAE` and `HiddenOutput` etc are all subclasses of `tf.Keras.Model` so we could probably just simplify the type annotations anyway... 

Will open a follow-up PR for this!